### PR TITLE
 Warm YUM RPMs cache in pre-restore-config event 

### PR DESCRIPTION
**Pull request by DavidePrincipi**
- download backup-config required packages
- download nethserver-dc required packages (if it is listed in backup-config)

The new `/etc/nethserver/ns6upgrade-package-list.sed` holds the packages conversion for the ns6 upgrade logic. It has been moved out of `restore-config-reinstall` to reuse it also in `restore-config-download`.

https://github.com/NethServer/dev/issues/5794

### CODE REVIEW

File: root/etc/e-smith/events/actions/restore-config-download [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# along with NethServer.  If not, see COPYING.
+#
+
+packagelist_path=/var/lib/nethserver/backup/package-list
+backup_path=/var/lib/nethserver/backup/backup-config.tar.xz
+
+packages=($(tar -x -O -f ${backup_path} ${packagelist_path:1} | sed -f /etc/nethserver/ns6upgrade-package-list.sed))
+
+if (( ${#packages[@]} == 0 )); then
+    echo "[WARNING] no packages listed for download"
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/restore-config-download:40]: I think this shouldn't be a WARNING but just a notice

File: root/etc/e-smith/events/actions/restore-config-reinstall [hunk 2]
```diff
@@ -19,21 +19,16 @@
...
-       -e 's/^nethserver-collectd-web$/nethserver-cgp/' \
-       -e 's/^nethserver-fetchmail$/nethserver-getmail/' \
-       -e 's/^nethserver-snort$/nethserver-suricata/' \
-       -e 's/^nethserver-ibays$/nethserver-virtualhosts\nnethserver-samba/' \
-       -e '/^nethserver-c-icap$/d' \
-       -e '/^nethserver-ipsec$/d' \
-    /var/lib/nethserver/backup/package-list
+# Migrate packages from NS 6 to NS 7
+# See: http://docs.nethserver.org/en/v7/release_notes.html#discontinued-packages
+sed -i -f /etc/nethserver/ns6upgrade-package-list.sed ${packagelist_path}
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/restore-config-reinstall:31]: If the `ns6upgrade-package-list.sed` file is not used somewhere else, I'd prefer to leave the code as is to improve readability.

File: root/etc/e-smith/events/actions/restore-config-download [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+packagelist_path=/var/lib/nethserver/backup/package-list
+backup_path=/var/lib/nethserver/backup/backup-config.tar.xz
+
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/restore-config-download:36]: This actions should download the packages only if `restore-config` is invoked without `no-reinstall`

File: root/etc/e-smith/events/actions/restore-config-download (hunk 1 above)
Comment (DavidePrincipi) [reply root/etc/e-smith/events/actions/restore-config-download:40]: I expect the list always contains some elements. If it is empty it is an abnormal condition and I want to raise it, though I can't consider it an error.

File: root/etc/e-smith/events/actions/restore-config-reinstall (hunk 2 above)
Comment (DavidePrincipi) [reply root/etc/e-smith/events/actions/restore-config-reinstall:31]: It is required by `restore-config-download`. I'd prefer not duplicating it!

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**